src/main/webapp/app/entities/programming-exercise/actions/programming-exercise-trigger-build-button.component.html

```diff
@@ -0,0 +1,9 @@
+<button
+    *ngIf="alwaysShowTriggerButton || (participationHasResult && participationIsActive && participationHasLatestSubmissionWithoutResult)"
```

Comment:
I am wondering if `participationHasResult` is really necessary here. What happens if `participationHasLatestSubmissionWithoutResult` evaluates to true, but there was no result yet?

Reply:
You are right, now that we only show the button when there is a failed submission, it doesn't make sense anymore. I removed this logic from the component.